### PR TITLE
fix: Issue with zoom meetings getting "invalid Meeting ID"

### LINKED ISSUE #16601: [CAL-4319] Issue with zoom meetings getting "invalid Meeting ID"
With some of the zoom links attached to [Cal.com](http://Cal.com) invites we have been getting the "Invalid meeting ID" error message at the moment of starting the meeting, and that's true for both participants, host and guest.

I was able to reproduce the issue in a test Cal.com account. I tested it with a brand new zoom app installed. The steps to reproduce the "invalid meeting id" issue are as follows:

 1. Have a Cal.com account configured with an event that has a Zoom app as location.
 2. Book an appointment using the link of that event.
 3. Login to the Cal.com account where the appointment was booked for in the above step.
 4. Go to Bookings
 5. Locate the recently scheduled appointment.
 6. Locate the menu on the right of the appointment details.
 7. Got to Edit -> Request reschedule.
 8. Wait for the reschedule to happen.
 9. Locate the new booking.
10. Click on the "Join Zoom Video" link.
11. Follow up Zoom's workflow to join the meeting.
12. You get the "Unable to join this meeting. This meeting ID is not valid" error message.

    

    

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-4319](https://linear.app/calcom/issue/CAL-4319/issue-with-zoom-meetings-getting-invalid-meeting-id)</sub>

### PR DESCRIPTION
**Pull request by askadityapandey**
## What does this PR do?

- Fixes #16601(GitHub issue number)
- Fixes CAL-[4319](https://linear.app/calcom/issue/CAL-4319/issue-with-zoom-meetings-getting-invalid-meeting-id)(Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
  - Ensure that the Zoom API credentials are correctly set up in the environment.
- What are the minimal test data to have?
  - Create a test appointment with a Zoom meeting.
  - Reschedule the appointment to trigger the update logic.
- What is expected (happy path) to have (input and output)?
  - The Zoom meeting ID should be correctly updated after rescheduling.
  - The updated meeting ID should be reflected in the appointment details.
- Any other important info that could help to test that PR
  - Ensure that the Zoom integration is enabled and properly configured in the Cal.com settings.

### CODE REVIEW

File: packages/app-store/zoomvideo/lib/VideoApiAdapter.ts [hunk 1]
```diff
@@ -333,12 +334,16 @@ const ZoomVideoApiAdapter = (credential: CredentialPayload): VideoApiAdapter =>
           body: JSON.stringify(await translateEvent(event)),
         });
 
-        return Promise.resolve({
+        // Fetch the updated meeting details
```
Comment (Amit91848) [inline packages/app-store/zoomvideo/lib/VideoApiAdapter.ts:337]: It is understood by the code, please remove the comment

File: packages/app-store/zoomvideo/lib/VideoApiAdapter.ts [hunk 2]
```diff
@@ -325,6 +325,7 @@ const ZoomVideoApiAdapter = (credential: CredentialPayload): VideoApiAdapter =>
     },
     updateMeeting: async (bookingRef: PartialReference, event: CalendarEvent): Promise<VideoCallData> => {
       try {
+        // Update the meeting details
```
Comment (Amit91848) [inline packages/app-store/zoomvideo/lib/VideoApiAdapter.ts:328]: Same here

File: packages/app-store/zoomvideo/lib/VideoApiAdapter.ts [hunk 3]
```diff
@@ -333,12 +334,16 @@ const ZoomVideoApiAdapter = (credential: CredentialPayload): VideoApiAdapter =>
...
+        return {
           type: "zoom_video",
-          id: bookingRef.meetingId as string,
-          password: bookingRef.meetingPassword as string,
-          url: bookingRef.meetingUrl as string,
-        });
+          id: result.id.toString(),
+          password: result.password || "",
+          url: result.join_url,
+        };
```
Comment (anikdhabal) [inline packages/app-store/zoomvideo/lib/VideoApiAdapter.ts:346]: Changes looks good, Pls a share a loom video of the result🙏

**Review by Amit91848 — APPROVED** [review]
LGTM! Tested and works 🚀🚀

Thank you for your contribution @askadityapandey

**Pull request merged**